### DISCUSSION TITLE
Is there any lib for composer that we could use for functionality that gave as XoopsForms?

### DISCUSSION THREAD

**Discussion by MekDrop** [ImpressCMS/impresscms#1025] [Q&A]
Maybe we could get rid of XoopsForms than?

**Comment by fiammybe** [comment]
Won't this be too fundamental a change for the 2.0 milestone? This is the library that is used throughout the core and the existing modules to build almost any form on the site.

**Comment by MekDrop** [comment]
I think most places now use ipf forms instead (and these forms extends xoops forms). But I'm not sure about modules. So that's why I'm asking about our options.

**Comment by fiammybe** [comment]
In that case, the title isn't 100% correct :-) I think removing xoopsforms would be a good step. But that would mean marking it as deprecated for 2.0 and removing it in 2.1. That spreads out the changes  that are needed in time, both for the core development as for people supporting modules (like me, @skenow @jegelstaff and @madfish)

**Comment by MekDrop** [comment]
I think I'll extract XoopsForms with all repo ristory into new composer package and place it in @OldXoopsLibraries and include this library in our core. It's possible that @geekwright and @mambax7 will move that package to @Xoops. They have a right to do so.

What I'm not sure about if we could use any other forms lib too for using in ipf forms part, so that they will not depend on XoopsForm. I think anything that doesn't use GPL license would be better much for us. Here I'm not sure if there are some alternatives. 

Another possibility - deprecate all forms PHP functionality and use [vue-form-generator](https://github.com/vue-generators/vue-form-generator) or similar library.

**Comment by MekDrop** [comment]
Probably we could use [symfony/form](https://github.com/symfony/form) for something to replace xoopsforms. 

One issue with this package is that it depends on [symfony/event-dispatcher](https://github.com/symfony/event-dispatcher), so for ImpressCMS/impresscms#177 we probably should use this package. Also, there is dependency for [symfony/intl](https://symfony.com/doc/current/components/intl.html)

**Reply by MekDrop** [reply]
Another alternative could be [laminas/laminas-form](https://docs.laminas.dev/laminas-form/).

**Reply by fiammybe** [reply]
What is the main problem you want to replace it? Licensing?
